src/engine/Engine.hpp

```diff
@@ -31,21 +31,14 @@ class Engine {
         const bool wasInserted = m_layers.emplace(idx, std::move(layer)).second;
 
         if (wasInserted && active) {
-            m_activeLayers.insert(idx);
+            m_deferredToggleLayers.push_back({idx, DeferType::Add});
```

Comment:
Why do we only handle activeLayers in the deferred Add?

Reply:
Because otherwise we would have to pass a reference to the added object. Also, since we cannot delte it until next frame anyways, it isnt't much of a problem.